left_sidebar: Standardize topic list filter input.

sayamsamal opened the pull request:
This follow-up PR replaces the current left sidebar topic list filter input implementation with the redesigned input_wrapper
component.

This PR also serves as the base for supporting inputs using the search_pill_widget, and thus adjusts the previously defined logic at certain places to ensure that the input pills are handled and displayed accurately.

Other changes in the PR:
- Fixes input color values deviating from the design.
- Stretch filter inputs in the left sidebar to fill available space on right. ([#issues > show all topics spacing @ 💬](https://chat.zulip.org/#narrow/channel/9-issues/topic/show.20all.20topics.20spacing/near/2215476))
- A prep commit moving the color variables of input pill to `app_variables.css`; away from `dark_theme.css`.
- Noticed some out of place selectors as an after effect of #34135, fixed all of them under one commit. (some of the changes here are for clarity purposes.)

Fixes part of #34476.

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

| Before | After |
|--------|--------|
| ![Screenshot 2025-07-10 at 6 26 18 AM](https://github.com/user-attachments/assets/3bfaef63-b726-45db-a3e0-26942fe1f3bf) | ![Screenshot 2025-07-10 at 6 25 44 AM](https://github.com/user-attachments/assets/af45e1a8-4bcd-490a-80c4-339eb03d219f) |
| ![Screenshot 2025-07-10 at 6 26 30 AM](https://github.com/user-attachments/assets/713b8b6e-d2a0-4cb8-ab88-ac205050f899) | ![Screenshot 2025-07-10 at 6 25 33 AM](https://github.com/user-attachments/assets/0afef01b-bb07-4914-9ec8-b2b05b464da1) | 

### In action
#### Light Mode

https://github.com/user-attachments/assets/8f583d5b-90fc-44c2-937b-5bc19aad52d7

#### Dark Mode

https://github.com/user-attachments/assets/322bc9d2-8a62-4763-beb5-50c72c1819c4

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

sayamsamal commented:
> 1. When I first open the "all topics view", there's an `x` in an empty filter area which I wouldn't expect to be there. Later on, there is no `x` if the filter is empty.

@alya would we want to show the `x` in the filter input when the `-unresolved` pill is active by default following #35143?

alya commented:
Yeah, I would expect an `x` whenever the field is not empty (has any pill or text).

sayamsamal commented:
### Update:
- Aligned the filter input horizontally with the rest of the elements on the left sidebar (fix was related to also considering the width of the borders)

| Normal Text | Input pill |
|--------|--------|
| <img width="861" height="293" alt="Screenshot 2025-07-12 at 1 10 00 PM" src="https://github.com/user-attachments/assets/6db25c78-c6ec-4711-9609-dba41fe15e2b" /> | <img width="861" height="293" alt="Screenshot 2025-07-12 at 1 10 57 PM" src="https://github.com/user-attachments/assets/fc851033-4508-4bf4-8de5-fbb3551a3c97" /> | 
- Fixed the logic for showing the close button in the filter inputs with CSS. Removed the previously used additional JS to keep track of empty/non-empty inpts as the current CSS implementation feels robust enough to handle all cases.

https://github.com/user-attachments/assets/0ca1a363-95b9-474b-99d0-349964dd1c91

sayamsamal commented:
@alya Fixed the issues in the feedback, this is ready for another review.

alya commented:
Looks good to go from my perspective!

timabbott commented on web/src/topic_list.ts:
```diff
@@ -494,7 +494,7 @@ export function zoom_in(): void {
 
 export function get_left_sidebar_topic_search_term(): string {
-    return $("#left-sidebar-filter-topic-input .input").text().trim();
+    return $("#topic_filter_query").text().trim();
```
In the future, commits like this should say in the body of the commit message what the bug was :)

timabbott commented on web/src/click_handlers.ts:
```diff
@@ -838,7 +838,7 @@ export function initialize(): void {
     // LEFT SIDEBAR
 
-    $("body").on("click", "#clear_search_topic_button", topic_list.clear_topic_search);
+    $("body").on("click", ".filter-topics .input-button", topic_list.clear_topic_search);
```
This is not a good selector -- previously, it was very clear that this was the "clear topic" button, now it's just "whatever input-button is inside this input".

We can fix this in a follow-up, but we should add a class to the close button that selectors like this can target.

timabbott commented on web/styles/inputs.css:
```diff
@@ -48,16 +48,18 @@
         padding-left: calc(
             var(--input-icon-starting-offset) + var(--input-icon-width) +
-                var(--input-icon-content-gap)
+                var(--input-icon-content-gap) - 1px
```
This complexity makes me wonder if we could use an `outline` rather than a border.

timabbott commented:
I found one bug, which is that if you type some stuff into the topics filter widget, and then Backspace it away, we get left with the "X" still visible.

<img width="830" height="470" alt="image" src="https://github.com/user-attachments/assets/41d91462-fff7-43f5-bb9f-16c0c02910b9" />

I think it's probably correct to merge this and fix that and the feedback above in a follow-up, since it'll simplify the rest of the work to polish this part of the project for the release. But please prioritize investigating that bug fix; I'm guessing it relates to the comment previously in `update_clear_button` about the invisible `<br>` tag.
